js/ui/date_box/ui.date_box.mask.js

```diff
@@ -34,6 +34,10 @@ var DateBoxMask = DateBoxBase.inherit({
         });
     },
 
+    _useMaskBehavior: function() {
+        return this.option("useMaskBehavior") && this.option("mode") === "text" && this.option("displayValue");
```

Comment:
Add a test which checks that the mask behavior is disabled when the displayValue option is undefined.